File: backend/src/Planty.Domain/Entities/Plant.cs
namespace Planty.Domain.Entities;

public class Plant
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public string Name { get; set; } = string.Empty;
    public string? Species { get; set; }
    public string? Description { get; set; }
    public DateTime DateAdded { get; set; } = DateTime.UtcNow;
    public int? WateringIntervalDays { get; set; }
    public string? ImageUrl { get; set; }

    // User relationship
    public Guid UserId { get; set; }
    public User User { get; set; } = null!;

    // Location relationship
    public Guid? LocationId { get; set; }
    public Location? Location { get; set; }

    // Watering history
    public ICollection<Watering> Waterings { get; set; } = new List<Watering>();
}

File: frontend/web/src/models/plant.js
/**
 * @typedef {Object} Plant
 * @property {string} id
 * @property {string} name
 * @property {string | null} species
 * @property {string | null} description
 * @property {string} dateAdded
 * @property {string | null} lastWatered
 * @property {number | null} wateringIntervalDays
 * @property {string | null} location
 * @property {string | null} imageUrl
 * @property {string | null} nextWateringDue
 * @property {string | null} lastFertilized
 * @property {number | null} fertilizationIntervalDays
 * @property {string | null} nextFertilizationDue
 */

/**
 * @typedef {Object} Watering
 * @property {string} id
 * @property {string} wateredAt
 * @property {string | null} notes
 */

/**
 * @typedef {Object} Fertilization
 * @property {string} id
 * @property {string} fertilizedAt
 * @property {string | null} notes
 */

/**
 * @typedef {Object} CreatePlantRequest
 * @property {string} name
 * @property {string | null} species
 * @property {string | null} description
 * @property {number | null} wateringIntervalDays
 * @property {number | null} fertilizationIntervalDays
 * @property {string | null} location
 */
